lib/internal/fs/promises.js

```diff
@@ -528,17 +527,24 @@ async function readFileHandle(filehandle, options) {
 
   let size = 0;
   let length = 0;
-  if ((statFields[1/* mode */] & S_IFMT) === S_IFREG) {
-    size = statFields[8/* size */];
+
+  if ((statFields[1 /* mode */] & S_IFMT) === S_IFREG) {
+    size = statFields[8 /* size */];
+
+    if (size > kIoMaxLength) {
+      process.emitWarning(
+        `Detected \`fs.readFile()\` to read a huge file in memory (${size} bytes). Please consider using \`fs.createReadStream()\` instead to minimize memory overhead and increase the performance.`,
```

Comment:
IMO instead of calling the file "huge", which can be interpreted as subjective, we should be explicit.

For example, `... a file larger than the recommended limit (${size} > ${limit})...`